### DISCUSSION TITLE
Extending /craft /ender /anvil with loom, furnace types, grindstone, etc.

### DISCUSSION THREAD

**Discussion by mrfloris** [Zrips/CMI#5021] [Feature Requests]
I've requested this in the past and we now have /anvil, which I am thankful for. But it sure would be fantastic if we have things like

```
/cmi furnace
/furnace
asConsole! cmi furnace [playerName]

Furnace type:
-default
-smoker
-blaster
-campfire

And fuel option:
-coal:12
-charcoal:15
-bucket (1x lava bucket)

(i understand you can put certain items in as fuel, but if players want that for all i care they use the normal furnace then)

And only use items from inventory:
-required

So we can make alias things like /blaster for vip group doing:
asConsole! cmi furnace [playerName] -blaster -coal:1 -required

Permissions could be through the custom aliases of course, but at the core
cmi.command.furnace (is also the default furnace type)
cmi.command.furnace.blaster
cmi.command.furnace.smoker
cmi.command.furnace.fuel
cmi.command.furnace.campfire
cmi.command.furnace.required.bypass
```

And the same for smithing table, grindstone, loom, and such.

Note: If the interface gets closed before it completes, it sounds logical that the task did not complete and the left over materials and/or completed materials drop on the floor or return to the inventory (if there's space, no space, drop on floor)

```
/cmi workbench
/cmi anvil
/cmi ender
/cmi backpack
/cmi enchantingtable
/cmi brewingstand
/cmi cauldron (as a perk to dye items and clean items)
/cmi loom
/cmi composter (so we can clear inventory into it to get something out of it)
/cmi cartography
/cmi fletching (maybe we can use this for something cool)
/cmi smithingtable
/cmi stonecutter
/cmi 
```

Thank you for considering.

**Comment by AZFire1** [comment]
Agreed. This would be useful to have..

**Comment by mrfloris** [comment]
This was a bug report ages ago, that got moved to a discussion, glad to see today's CMI release 9.0.4.0 to include some of these changes.

**Comment by mrfloris** [comment]
I believe we now have:
```
/cmi smithingtable
/cmi workbench
/cmi anvil
/cmi ender
- backpack via shulker (no command)
/cmi loom
/cmi cartographytable
/cmi smithingtable
/cmi stonecutter
```
No furnace, smoker, blaster, enchantingtable, campfire, brewingstand, cauldron, or the others. 

But we got some at least, thank you for adding.

**Comment by mrfloris** [comment]
I see another thread with this topic https://github.com/Zrips/CMI/discussions/3980

**Comment by CloudeLecaw** [comment]
didnt we had this theme on discord and also analysed and philosophized with the others there why might not be possible?
#virtual tools #virtual storeage #simulation of toolfunction

**Comment by MrCrazys** [comment]
It would likely be too hard to keep track of. For /cmi furnace, it would most probably just open a new Furnace each time, meaning that if you put something in the Furnace and close it, it would be cancelled (just going off of what I think would happen). It would most likely be very resource intensive to keep track of items in each player’s Furnace. I personally don’t think this is needed.
